task5/src/main/java/com/clouway/adapter/http/LoginController.java

```diff
@@ -31,24 +34,24 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
     String password = request.getParameter("password");
 
     if (userValidator.isValid(email, password)) {
-      authorize(email, password, response);
+      authenticate(email, password, response);
     } else {
       response.sendRedirect("/login?errorMsg=<h2 style='color:red'>Username or password invalid format!</h2>");
     }
   }
 
-  private void authorize(String email, String password, HttpServletResponse response) throws IOException, ServletException {
-    User user = userRepository.findByEmail(email);
-    if (user != null) {
-      if (user.getPassword().equals(password)) {
-        Cookie cookie = new Cookie("sessionId", email);
-        response.addCookie(cookie);
-        response.sendRedirect("/welcomehomepage");
-      } else {
-        response.sendRedirect("/login?errorMsg=<h2 style='color:red'>Wrong password!</h2");
-      }
+  private void authenticate(String email, String password, HttpServletResponse response) throws IOException, ServletException {
+    boolean isAuthenticated = userRepository.authorize(email, password);
+    if (isAuthenticated) {
```

Comment:
If remove the private `authenticate()` there will be composite `if` /if in if/ :
```java
if (validator.validate){
    if (userRepository.authenicate()){
        someLogic...
    }
}
``` 
It is not complicated to understand the logic this way but it can be easily escaped with something like:
```java
boolean valid = validator.validate();
if (!valid){
    return error message
}

boolean authentikated = userRepository.authenticate();
if (authenticated){
    do something...
}
``` 

If you apply this sugggestion you will be able to remove the private method